lib/Consul/API/Agent.pm

```diff
@@ -77,6 +81,10 @@ sub force_leave {
 sub check_register {
     my ($self, $check, %args) = @_;
     croak 'usage: $agent->check_register($check, [%args])' if grep { !defined } ($check);
+    {
+        local $Carp::CarpInternal{ (__PACKAGE__) } = 1;
```

Comment:
Why? I understand what `%Carp::CarpInternal` is for, but why do you use it here?